# TNPM Pipeline

Documentations for the bioinformatic analysis pipeline of the Tumor Neoantigen-based Personalized Medicine (TNPM) Project.

## Prepare the environment

We use container technology to assure the consistency of running environments. The stardard environment is described by the *Dockerfile* located in the root of this repository.

To build and run the environment:

```
sudo docker build --tag tnpm:latest --rm /PATH/TO/pipeline-TNPM
sudo docker run --name tnpm --volume /PATH/TO/WORKING_DIR:/data -it tnpm
```

For more info on docker, please refer to: https://docs.docker.com/engine/docker-overview/

## Index human genomes

We choose STAR as our aligner due to its robust performance with default parameters on different situations.

The RefSeq format of GRCh38.p12 was downloaded from NCBI and indexed by STAR:

```
# Decompress data files
gzip --decompress --stdout GCF_000001405.38_GRCh38.p12_genomic.fna.gz > GRCh38.p12.fa
gzip --decompress --stdout GCF_000001405.38_GRCh38.p12_genomic.gff.gz > GRCh38.p12.gff

# Convert gff3 to gtf
gffread-0.9.12 GRCh38.p12.gff -T -o GRCh38.p12.gtf

# Generate genome indexes
mkdir GRCh38.p12-STAR && cd GRCh38.p12-STAR
STAR --runThreadN 8 --runMode genomeGenerate \
  --genomeDir . --genomeFastaFiles ../GRCh38.p12.fa --sjdbGTFfile ../GRCh38.p12.gtf
```

There are two notes for this section:
1. Human genome indexes generated by STAR with default parameters will require 30G memory to load when mapping reads.
2. GRCh38.p12 (Genome Reference Consortium Human Build 38 patch release 12) is available in both GenBank (GCA) and RefSeq (GCF) format. Here we choose RefSeq over GenBank for no reason at present.

## Align reads

We use STAR to map reads and mark duplicates (important in removing PCR duplicates which could introduce bias in variant calling):

```
# Align reads
STAR --runThreadN 8 --runMode alignReads \
  --outSAMtype BAM SortedByCoordinate --outFilterScoreMinOverLread 0.3 --outFilterMatchNminOverLread 0.3 \
  --genomeDir genomes/GRCh38.p12-STAR --outFileNamePrefix xxx_ --readFilesIn xxx_1.fastq xxx_2.fastq

# Mark duplicates
STAR --runThreadN 8 --runMode inputAlignmentsFromBAM \
  --bamRemoveDuplicatesType UniqueIdentical --outFileNamePrefix xxx_ --inputBAMfile xxx.bam
```

## Call variants

We use GATK from the Broad Institute to call somatic variants:

```
gatk Mutect2 -R reference.fa \
  -I tumor.bam -tumor tumor_sample_name \
  -I normal.bam -normal normal_sample_name \
  -O somatic.vcf.gz
```
