# Running Guide

## Setup the environment

We use container technology to assure the consistency of running environments.

To build and run the environment:

```
sudo docker build --tag tnpm:latest --rm /PATH/TO/pipeline-TNPM
sudo docker run --name tnpm --volume /PATH/TO/WORKING_DIR:/data -it tnpm
```

For more info on docker, please refer to: https://docs.docker.com/engine/docker-overview/

## Prepare references

We choose the RefSeq format of GRCh38.p12 which was downloaded from NCBI as our references.

To get the reference genomes prepared:

```
# Decompress data files
gzip --decompress --stdout GCF_000001405.38_GRCh38.p12_genomic.fna.gz > GRCh38.p12.fa
gzip --decompress --stdout GCF_000001405.38_GRCh38.p12_genomic.gff.gz > GRCh38.p12.gff

# Prepare references for STAR
gffread GRCh38.p12.gff -T -o GRCh38.p12.gtf
mkdir GRCh38.p12-STAR
STAR --runThreadN 8 --runMode genomeGenerate \
  --genomeDir ./GRCh38.p12-STAR --genomeFastaFiles GRCh38.p12.fa --sjdbGTFfile GRCh38.p12.gtf

# Prepare references for GATK
samtools faidx GRCh38.p12.fa
gatk CreateSequenceDictionary -R GRCh38.p12.fa -O GRCh38.p12.dict
```

There are two notes for this section:
1. Human genome indexes generated by STAR with default parameters will require 30G memory to load when mapping reads.
2. GRCh38.p12 (Genome Reference Consortium Human Build 38 patch release 12) is available in both GenBank (GCA) and RefSeq (GCF) format. Here we choose RefSeq over GenBank for no reason at present.

## Align reads

We choose STAR as our aligner due to its robust performance with default parameters on different situations.

To map reads and mark duplicates (important in removing PCR duplicates which could introduce bias in variant calling):

```
# Align reads
STAR --runThreadN 8 --runMode alignReads \
  --outSAMtype BAM SortedByCoordinate --outSAMmapqUnique 60 \
  --outFilterScoreMinOverLread 0.3 --outFilterMatchNminOverLread 0.3 \
  --genomeDir genomes/GRCh38.p12-STAR \
  --outFileNamePrefix xxx_ --readFilesIn xxx_1.fastq xxx_2.fastq

# Mark duplicates
STAR --runThreadN 8 --runMode inputAlignmentsFromBAM \
  --bamRemoveDuplicatesType UniqueIdentical \
  --outFileNamePrefix xxx_ --inputBAMfile xxx_Aligned.sortedByCoord.out.bam.bam

# Prepare BAM for GATK
gatk AddOrReplaceReadGroups -LB lib -PL illumina -PU unit \
  -SM xxx -I xxx_Processed.out.bam -O xxx.bam \
samtools index xxx.bam
```

## Call variants

To use GATK from the Broad Institute to call somatic variants:

```
gatk Mutect2 -R genomes/GRCh38.p12.fa \
  -I xxx1.bam -tumor xxx1 -I xxx2.bam -normal xxx2 -O somatic.vcf.gz
```
